static/app/components/platformPicker.tsx

```diff
@@ -233,7 +229,16 @@ function PlatformPicker({
                     source,
                     organization: organization ?? null,
                   });
-                  setPlatform({...item, category});
+
+                  const itemCategories = categories
+                    .filter(cat => cat.platforms.has(item.id))
+                    .map(cat => cat.id);
+
+                  if (itemCategories.includes(category)) {
+                    setPlatform({...item, category});
+                  } else {
+                    setPlatform({...item, category: itemCategories[0] ?? 'all'});
+                  }
```

Comment:
this is when for example, the "server" tab is selected but the user goes and filter "react". When going back if the     "server" tab is selected, the selected platform "react" is not shown, it should goes to where we have "react" listed.